frontend/src/components/Permissions.tsx

```diff
@@ -1,50 +1,119 @@
-import { PlusCircle } from "lucide-react";
-import React, { useEffect, useState } from "react";
+import { format } from "date-fns";
+import { CalendarIcon, PlusCircle, XIcon } from "lucide-react";
+import React, { useState } from "react";
+import Scopes from "src/components/Scopes";
 import { Button } from "src/components/ui/button";
-import { Checkbox } from "src/components/ui/checkbox";
+import { Calendar } from "src/components/ui/calendar";
+import { Input } from "src/components/ui/input";
 import { Label } from "src/components/ui/label";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "src/components/ui/popover";
 import {
   Select,
   SelectContent,
   SelectItem,
   SelectTrigger,
   SelectValue,
 } from "src/components/ui/select";
-import { useCapabilities } from "src/hooks/useCapabilities";
 import { cn } from "src/lib/utils";
 import {
   AppPermissions,
   BudgetRenewalType,
+  NIP_47_PAY_INVOICE_METHOD,
   Scope,
+  WalletCapabilities,
   budgetOptions,
   expiryOptions,
   iconMap,
   scopeDescriptions,
   validBudgetRenewals,
 } from "src/types";
 
+const getTimeZoneDirection = () => {
```

Comment:
This is because the Date Picker component gives 00:00 <your time zone> date. And we deal with UTC date. So we have to +/- day depending on the time zone direction. I still feel expiring the app on 23:59 of the time zone of the user is more sensible than UTC 23:59